dnf-behave-tests/features/repoquery/rich-deps.feature

```diff
@@ -0,0 +1,187 @@
+Feature: Tests for rich (boolean) dependencies:
+ and - requires all operands to be fulfilled for the term to be True.
+ or - requires one of the operands to be fulfilled
+ if - requires the first operand to be fulfilled if the second is (reverse implication)
+ if else - same as above but requires the third operand to be fulfilled if the second is not
+ with - requires all operands to be fulfilled by the same package for the term to be True.
+ without - requires a single package that satisfies the first operand but not the second (set subtraction)
+ unless - requires the first operand to be fulfilled if the second is not (reverse negative implication)
+ unless else - same as above but requires the third operand to be fulfilled if the second is
+
+Background:
+Given I use the repository "repoquery-rich-deps"
+
+
+Scenario: repoquery --recommends NAME
+ When I execute dnf with args "repoquery --recommends c1"
+ Then the exit code is 0
+  And stdout is
+      """
+      (b1 and a1)
+      """
+
+
+# c1-1.0: Requires: (a1-prov1 if b1)
+Scenario: repoquery --whatrequires for "(a1-prov1 if b1)"
+ When I execute dnf with args "repoquery --whatrequires a1"
+ Then the exit code is 0
+  And stdout is
+      """
+      c1-0:1.0-1.x86_64
+      c1-0:2.0-1.x86_64
+      """
+
+
+# a1-1.0: Conflicts: ((b1 and x1) or c1)
+Scenario: repoquery --whatconflicts for "((b1 and x1) or c1)"
```

Comment:
Please specify type of arguments - provide or package name?

Reply:
Every package provides its name and I assume these commands work with provides, so they are provides, in this case the implicit ones. I don't think there's anything unclear here, but if you do think so, let me know.